lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -193,6 +213,23 @@ export class ClientCredentialClient extends BaseClient {
                 headers,
                 thumbprint
             );
+
+            // check if a new token is being requested because the cached token needs to be refreshed, not because it's expired
+            if (accessTokenCacheKey) {
+                // check if ESTS is down (status code 429 or 5xx)
+                if ((response.status === 429) || (response.status >= 500)) {
+                    this.logger.warning(
+                        "ClientCredentialClient:executeTokenRequest - AAD is currently unavailable and is unable to refresh the token."
+                    );
+                // ESTS is not down, the cached token is bad and should be removed from the cache
+                } else {
+                    this.logger.error(
+                        "ClientCredentialClient:executeTokenRequest - AAD is currently available but is unable to refresh the token. Returning the unexpired token and removing it from the cache."
+                    );
+                    this.cacheManager.removeAccessToken(accessTokenCacheKey);
```

Comment:
I can't think of a scenario where you'd want to remove an access token. Generally MSAL relies on access tokens being replaced.

Reply:
I agree, but I see this bullet in the long term solution in the 'refresh access token' spec: ```If MsalServiceException is thrown and the AAD is available, log the error and remove the entry from the cache```. It appears the token will have already been replaced anyways because of how everything currently functions. I can remove this. Can you confirm?

Reply:
Let's to skip this part as it can have unintended consequences. Removing tokens from cache is not simple (particularly for user flows, because you need to remove RT and IDT and Accounts as well). 

If AAD is available and it issues a 4xx error (e.g. cert expired), the error bubbles up to the app. That's a good enough signal.

Reply:
(feel free to change the spec on this one)